doc/sourmash-internals.md

```diff
@@ -0,0 +1,638 @@
+# A guide to the internals of sourmash
+
+```{contents} Contents
+:depth: 3
+```
+
+sourmash was created in 2015, and has been repeatedly reorganized,
+refactored, and optimized to support ever larger databases, faster
+queries, and new use cases. We've also regularly added new
+functionality and features.  So sourmash can be pretty complicated
+internally, and our user-facing documentation only covers a fraction
+of its potential!
+
+This document is a brain dump intended for expert users and sourmash
+developers who want to understand how, why, and when to use various
+sourmash features. It is unlikely ever to be comprehensive, so the
+information you are interested in may not yet exist in this document,
+but we are always happy to add to it -
+[just ask in an issue!](https://github.com/sourmash-bio/sourmash/issues)
+
+## Signatures and sketches
+
+sourmash operates on sketches. Each sketch is a collection of hashes.
```

Comment:
```suggestion
sourmash operates on sketches. Each sketch is a collection of hashes. Each hash is a whole number product of a k-mer and hash function. 
```
Maybe link the hash function to the ## making sketches section of the document. May wish to include another line about k-mers as well. Such as, 
"A k-mer is a short sub-sequence generated from a larger sequence file via a sliding window approach. 

I.e. If k=3, a 3-mer of an 8 base pair DNA molecule will have 6 individual 3-mers digested into hashes to populate a  sketch that are stored in a signature.

> The DNA sequence:
> AGTCATCG

> The k-mers of the sliding window process:
> [AGT].....
> .[GTC]....
> ..[TCA]...
> ...[CAT]..
> ....[ATC].
> .....[TCG]

> Hashes digested from each k-mer:
> 2118360698
> 1681319365
> 65865673
> 1255238040
> 364627471
> 659934874
"

Reply:
we could point at https://sourmash.readthedocs.io/en/latest/kmers-and-minhash.html for a basic intro, rather than rehashing (hah!) k-mers. What do you think?

Here are my revisions to the intro para - thoughts?

>sourmash operates on sketches. Each sketch is a collection of hashes,
which are in turn built from k-mers by appling a hash function
(currently always murmurhash) and a filtering function.  Each sketch
is contained in a signature wrapper that contains some metadata.